addons/crm/tests/test_crm_lead_multicompany.py

```diff
@@ -62,6 +62,41 @@ def test_lead_mc_company_computation(self):
         self.assertEqual(lead_team_no_company.company_id, self.company_2)
         self.assertEqual(lead_team_no_company.team_id, self.team_company2)
 
+    @users('user_sales_manager_mc')
+    def test_lead_mc_company_computation_with_restricted_allowed_company_ids(self):
```

Comment:
> As what we test is that is that team company wins over environment companies, isn't it ?

Yes, it is in order to avoid inconsistencies between the team_company and the company_id.